### PR TITLE
`phylum auth set-token`

### PR DESCRIPTION
**Pull request by kylewillmon**
This patch adds a command to set the token in the config file. This provides an alternative to `phylum auth login` provided you can get the token from some other place. The other place may be `phylum auth token` from another machine or, in the future, we may support copying the token directly from the web UI.

### CODE REVIEW

**Review by furi0us333 — CHANGES_REQUESTED** [review]
We should update the docs for this new subcommand

File: cli/src/commands/auth.rs [hunk 1]
```diff
@@ -88,28 +109,27 @@ pub async fn handle_auth(
     app_helper: &mut Command,
     timeout: Option<u64>,
 ) -> CommandResult {
-    if matches.subcommand_matches("register").is_some() {
-        match handle_auth_register(config, config_path).await {
+    match matches.subcommand() {
```
Comment (andreaphylum) [inline cli/src/commands/auth.rs:112]: I always like it when `if` chains become `match` blocks. 💯

**Review by andreaphylum — APPROVED** [review]

**Review by maxrake — APPROVED** [review]
The behavior was confirmed with a local build and tested with both a token provided as an option and one provided from stdin. Other than adding documentation, LGTM.

**Review by maxrake — APPROVED** [review]

**Review by furi0us333 — APPROVED** [review]
LGTM

File: docs/command_line_tool/phylum_auth.md [hunk 2]
```diff
@@ -9,5 +9,6 @@ Manage authentication, registration, and API keys
 ### Commands
 * [phylum auth login](https://docs.phylum.io/docs/phylum_auth_login)
 * [phylum auth register](https://docs.phylum.io/docs/phylum_auth_register)
+* [phylum auth set-token](https://docs.phylum.io/docs/phylum_auth_set-token)
```
Comment (maxrake) [inline docs/command_line_tool/phylum_auth.md:12]:
```suggestion
* [phylum auth set-token](https://docs.phylum.io/docs/phylum_auth_set_token)
```
The other possibility is to change the file name to `docs/command_line_tool/phylum_auth_set-token.md`...which would match the other command/page with a hyphenated name: `docs/command_line_tool/phylum_project_set-thresholds.md`

The working assumption here is the file name is what creates the page URL/slug.

**Review by maxrake — CHANGES_REQUESTED** [review]
The new link or the new doc filename needs to change.

**Review by maxrake — APPROVED** [review]

**Review by cd-work — DISMISSED** [review]

**Review by cd-work — APPROVED** [review]

**Pull request merged**